cdap-app-fabric/src/main/java/co/cask/cdap/app/mapreduce/MRJobClient.java

```diff
@@ -82,6 +82,10 @@ public MRJobInfo getMRJobInfo(Id.Run runId) throws IOException, NotFoundExceptio
     JobStatus thisJob = findJobForRunId(jobs, runId);
 
     RunningJob runningJob = jobClient.getJob(thisJob.getJobID());
+    if (runningJob == null) {
+      LOG.warn("JobClient returned null for JobId: '{}'", thisJob.getJobID());
+      throw new NotFoundException(runId);
```

Comment:
But if we `throw new NotFoundException(runId)`, the message that user will get is "runid not found", which would be incorrect since the runid exists in the system. Maybe throw an exception with more accurate message.